### ISSUE TITLE
Update HTTPS Certificate After Hostname Changes

### ISSUE THREAD

**Issue by NickDaly** [freedombox/Plinth#598]
When the user changes the hostname via `plinth/sys/config`, the certificate subject's common name (CN) doesn't change from "freedombox" to the new name.  The user is forced to add a security exception for their FreedomBox to connect, which should be avoided if at all possible.

Additionally, the default certificate's CN is incorrect: it uses the "freedombox" CN, while the user is connecting to "freedombox.local".

**TODOs:**

- [ ] Change default certificate CN to "freedombox.local" instead of "freedombox".
- [ ] Update certificate's CN to new hostname after user updates hostname.

**Comment by SunilMohanAdapa** [comment]
On 09/11/2016 03:10 AM, Nick Daly wrote:
> When the user changes the hostname via |plinth/sys/config|, the
> certificate subject's common name (CN) doesn't change from "freedombox"
> to the new name. The user is forced to add a security exception for
> their FreedomBox to connect, which should be avoided if at all possible.

The security exception will need to be added anyway because the
certificate is a self signed one, right?

Also, I think hostname of the system has nothing to do with the
certificate.  It should be domain name (of which there can be many
pointing to the box) that should be part of the CN.  Currently, if the
domain name changes, one can go to let's encrypt and obtain a new
certificate for that domain.  At this point new web server configuration
is created for the domain to use the new certificate.

> 
> Additionally, the default certificate's CN is incorrect: it uses the
> "freedombox" CN, while the user is connecting to "freedombox.local".
> 

This is for initially generated snake-oil certificate?

[...]

-- 
Sunil

**Comment by NickDaly** [comment]
> The security exception will need to be added anyway because the certificate is a self signed one, right?

Correct.  The user could decide to trust their freedombox as a CA, but that's asking too much of our target user-base.  Fixing this would reduce the number of things wrong with the cert to 1 instead of 2.  I've corrected the top comment.

> This is for initially generated snake-oil certificate?

Also correct.
